### ISSUE TITLE
Codelite crashed and deleted itself and files

### ISSUE THREAD

**Issue by buttsy83** [eranif/codelite#2107]
So I was just working on Codelite and ran a debugging session. Codelite then decided to close down and delete itself, along with the cpp file I have been working on for the last 2 weeks.
Am I able to get this file back?

**Comment by buttsy83** [comment]
I have the cpp.o and the cpp.o.d file, am I able to convert these back to the cpp file?

**Comment by eranif** [comment]
which version of codelite? 
which OS?
usually,  CodeLite writes to a temporary files and then rename the temp file.  this way,  if CodeLite crashes during file wire,  the two file should still be available.

**Comment by buttsy83** [comment]
Hey, thanks for the quick response.

Pretty sure it was version 12 of the software. Windows 10.

I’ve noticed in the debug folder of the current work folder, it has made a cpp.o and cpp.o.d file. Are these able to be converted back to cpp?

Cheers,

Matt

Sent from Mail for Windows 10

**Comment by buttsy83** [comment]
Pretty sure it was version 12 of the software. Windows 10.

I’ve noticed in the debug folder of the current work folder, it has made a cpp.o and cpp.o.d file. Are these able to be converted back to cpp?

**Comment by buttsy83** [comment]
where does it store the temp files?

**Comment by eranif** [comment]
The file name is located on the same folder as your code.
It starts with `~` followed by the file name.

This will help you only if CodeLite crashed during file writing.

**Comment by buttsy83** [comment]
Unfortunately no temp file. Only the .mk .project .o .o.d files. Its weird because I had just started debugging using the .cpp file along with 2 other class files (2 x .cpp + 2 x .h) files, and only the source .cpp has been deleted, along with the whole Codelite itself.

No chance of converting the .o file back to .cpp?

**Comment by eranif** [comment]
No, if you can convert it back to C++ - you should sell it, you will do alot of money out of it :)
Sorry about your source file, shit happens :/
 try setting up a local source control for exactly such cases
